### PR TITLE
Add Histogram Sum Aggregation

### PR DESCRIPTION
**Pull request by zeotuan**
Implement Sum Aggregation as Part of #31

### CODE REVIEW

**Review by zeotuan — COMMENTED** [review]
@SemyonSinchenko Should we remove these generated  protobuf class from tsumugi-python

**Comment by SemyonSinchenko** [comment]
@zeotuan How did you generate it? There [is](https://github.com/SemyonSinchenko/tsumugi-spark/blob/main/Makefile#L5) a `make generate_code` that also apply linting rules and with it diff should be less.

File: tsumugi-python/tsumugi/analyzers.py [hunk 1]
```diff
@@ -33,11 +32,29 @@ class FilteredRowOutcome(Enum):
...
 
-class AggregateFunction(Enum):
-    """Histogram aggregation functions."""
+class AbstractAggregateFunction(ABC):
+    """Abstract class Histogram aggregation functions."""
 
-    COUNT = proto.Histogram.AggregateFunction.Count
-    SUM = proto.Histogram.AggregateFunction.Sum
+    @abstractmethod
+    def to_proto_aggregate(self) -> proto.Histogram.AggregateFunction: ...
```
Comment (SemyonSinchenko) [inline tsumugi-python/tsumugi/analyzers.py:39]: Let's name it `_to_proto` instead like in all other places?

File: tsumugi-server/src/main/scala/com/ssinchenko/tsumugi/DeequSuiteBuilder.scala [hunk 2]
```diff
@@ -141,9 +141,10 @@ object DeequSuiteBuilder {
           if (protoAnalyzer.hasWhere) Some(protoAnalyzer.getWhere) else Option.empty,
           if (protoAnalyzer.hasComputeFrequenciesAsRatio) protoAnalyzer.getComputeFrequenciesAsRatio else true,
           if (protoAnalyzer.hasAggregateFunction) {
-            protoAnalyzer.getAggregateFunction match {
-              case proto.Histogram.AggregateFunction.Count => Histogram.Count
-              case proto.Histogram.AggregateFunction.Sum   => throw new RuntimeException("Sum is not supported yet")
+            val protoAggregate = protoAnalyzer.getAggregateFunction
+            protoAggregate.getAggregateFunctionCase match {
+              case proto.Histogram.AggregateFunction.AggregateFunctionCase.COUNT_AGGREGATE => Histogram.Count
+              case proto.Histogram.AggregateFunction.AggregateFunctionCase.SUM_AGGREGATE  => Histogram.Sum(protoAggregate.getSumAggregate.getAggColumn)
```
Comment (SemyonSinchenko) [inline tsumugi-server/src/main/scala/com/ssinchenko/tsumugi/DeequSuiteBuilder.scala:147]: Can we add also a simple test for that case?

File: tsumugi-server/src/main/scala/com/ssinchenko/tsumugi/DeequSuiteBuilder.scala [hunk 3]
```diff
@@ -141,9 +141,10 @@ object DeequSuiteBuilder {
...
               case _ => throw new RuntimeException("Unknown AggregateFunction type!")
               // TODO: it is possible but tricky to implement the support of Sum; good first issue
```
Comment (SemyonSinchenko) [inline tsumugi-server/src/main/scala/com/ssinchenko/tsumugi/DeequSuiteBuilder.scala:149]:
```suggestion
```

File: tsumugi-python/tsumugi/analyzers.py [hunk 4]
```diff
@@ -1,4 +1,3 @@
-# Docstring of classes were heavily inspired by the
```
Comment (SemyonSinchenko) [inline tsumugi-python/tsumugi/analyzers.py:1]: Why is the line marked as deleted?

Comment (SemyonSinchenko) [inline tsumugi-python/tsumugi/analyzers.py:1]: Why the line is deleted?

**Review by SemyonSinchenko — APPROVED** [review]
@zeotuan Thanks a lot for the contribution! LGTM overall, I left to minor things about comments in the code.

File: tsumugi-python/tsumugi/analyzers.py (hunk 4 above)
Comment (zeotuan) [reply tsumugi-python/tsumugi/analyzers.py]: Restored

**Pull request merged**